content/doc/book/system-administration/administering-jenkins-on-kubernetes.adoc

```diff
@@ -12,4 +12,176 @@ ifdef::backend-html5[]
+
+* Cross-check the list of deprecations you find using the list from link:https://kubernetes.io/docs/reference/using-api/deprecation-guide[Kubernetes], and see what is required for the Kubernetes version you want to upgrade to.
+
+* Fix what needs to be fixed, and deploy the changes.
+
+You may now start the upgrade of Kubernetes.
+The process has two steps: 
+. Upgrade the control plane.
+. Upgrade the nodes. 
+
+Follow the upgrade steps according to your cloud provider.
+* link:https://docs.aws.amazon.com/eks/latest/userguide/update-cluster.html[Amazon]
+* link:https://learn.microsoft.com/en-us/azure/aks/upgrade-cluster?tabs=azure-cli[Azure]
+* link:https://cloud.google.com/kubernetes-engine/docs/how-to/upgrading-a-cluster[Google]
+* Pour yourself a cup of your favorite drink while you sit back and wait for the magic to happen.
+
+=== Upgrade Kubernetes Cluster maintained by the Jenkins Infrastructure project:
+
+*Requirements:*
+
+* Create a link:https://issues.jenkins.io[Jira] ticket.
+
+Announce on:
+* link:https://github.com/jenkins-infra/status[Jenkins Status] page.
+* IRC, channel #jenkins-infra on freenode.
+* link:https://groups.google.com/g/jenkins-infra[Jenkins Infrastructure and Dev Mailing list]
+* link:https://twitter.com/jenkinsci/[Twitter], in case of major upgrade.
+
+Review Release Changelogs:
+* link:https://github.com/kubernetes/kubernetes/tree/master/CHANGELOG[Kubernetes]
+* link:https://github.com/Azure/AKS/blob/master/CHANGELOG.md[AKS]
+
+Review and merge important pull request from link:https://github.com/jenkins-infra/charts[jenkins-infra/charts].
+
+Block k8smgmnt job on infra.ci
+
+Pagerduty
+* Bump kubectl docker image link:https://github.com/jenkins-infra/docker-helmfile[docker-helmfile] to the targeted kubernetes controller version.
+* Apply in the jenkins-infra/charts repo: https://github.com/jenkins-infra/charts/pull/1056.
+
+Search for deprecated Kubernetes resources.
+* link:https://github.com/FairwindsOps/pluto[Pluto] running 'pluto detect-helm -owide'
+
+NOTE: If you want to learn more about upgrades please link:https://github.com/jenkins-infra/documentation/tree/main/maintenance/kubernetes[refer to this link]!
+
+== Upgrade Jenkins Inside a Docker Container:
+
+First identify your image.
+
+----
+
+$ docker ps --format "{{.ID}}: {{.Image}} {{.Names}}"
+3d2fb2ab2ca5: jenkins-docker jenkins-docker_1
+
+----
+
+Then login into the image as root.
+
+----
+
+$ docker container exec -u 0 -it jenkins-docker_1 /bin/bash
+
+----
+
+Now you are inside the container, download the 'jenkins.war' file from the official site link.
+
+----
+
+# wget wget http://updates.jenkins-ci.org/download/war/2.176.1/jenkins.war
+
+----
+
+Replace the version with the one that fits to you.
+
+The next step is to move that file and replace the oldest one.
+
+----
+
+# mv ./jenkins.war /usr/share/jenkins/
+
+----
+
+Then change permissions.
+
+----
+
+# chown jenkins:jenkins /usr/share/jenkins/jenkins.war
+
+----
```

Comment:
Administrators should never, ever upgrade Jenkins from **inside** the container.  Software packages (like Jenkins) inside a container image should be considered immutable and should be treated as though they are immutable.  When it is time to upgrade Jenkins, it is time to prepare a new container image, pull the new container image into Kubernetes, stop the running container image, and start the new container image.

Please remove this section from the document.

Can you point me to the location where you found this recommendation?  We need to alert those people that this is a poor choice and leads to configuration management issues in the Jenkins configuration.
```suggestion
```

Reply:
I found this information while browsing the internet for making this document. 
https://gist.github.com/ricardo-dlc/1a6cbd9d48e4fa791b022b1eebc15de5